### ISSUE TITLE
Bug: Unable to remove interface defined header value by nullifying on method level

### ISSUE THREAD

**Issue by jpe-trustpilot** [canton7/RestEase#29]
According to the [documentation](https://github.com/canton7/RestEase#redefining-headers) you should be able to remove a header by redefinition.

```
[Header("X-InterfaceAndMethod-ToBeRemoved", "InterfaceValue")]
public interface ISomeApi
{
  ...
  [Header("X-InterfaceAndMethod-ToBeRemoved", null)]
  Task DoSomethingAsync(
    ...
  )
}
```

However, having code like this, results in an exception. Below is the output from the attached POC project
```
Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: str
   at System.Reflection.Emit.ModuleBuilder.GetStringConstant(String str)
   at System.Reflection.Emit.ILGenerator.Emit(OpCode opcode, String str)
   at RestEase.Implementation.ImplementationBuilder.AddMethodHeader(ILGenerator methodIlGenerator, HeaderAttribute header)
   at RestEase.Implementation.ImplementationBuilder.AddMethodHeaders(ILGenerator methodIlGenerator, MethodInfo methodInfo)
   at RestEase.Implementation.ImplementationBuilder.HandleMethods(TypeBuilder typeBuilder, Type interfaceType, FieldBuilder requesterField, FieldInfo classHeadersField, AllowAnyStatusCodeAttribute classAllowAnyStatusCodeAttribute, SerializationMethodsAttribute classSerializationMethodsAttribute, PropertyGrouping properties)
   at RestEase.Implementation.ImplementationBuilder.BuildImplementationImpl(Type interfaceType)
   at RestEase.Implementation.ImplementationBuilder.CreateImplementation[T](IRequester requester)
   at RestEasePOC.Program.Main(String[] args) in D:\POC\RestEasePOC\RestEasePOC\Program.cs:line 11

```

The ability to remove a "default" header for certain endpoints are very convenient.

[RestEasePOC.zip](https://github.com/canton7/RestEase/files/837905/RestEasePOC.zip)

**Comment by canton7** [comment]
Right, that should be fixed. I'll try and release over lunch.

**Comment by jpe-trustpilot** [comment]
Thank you for your prompt response / fix. 

I have run my POC project towards the develop branch and can confirm that it solves the problem.

Looking forward for the nuget release. Do you have any guesstimate to when that might be?

**Comment by canton7** [comment]
In the next hour and a half, hopefully.

**Comment by canton7** [comment]
Righty, released as 1.3.3
